sdl_android/src/main/java/com/smartdevicelink/api/audio/BaseAudioDecoder.java

```diff
@@ -0,0 +1,226 @@
+package com.smartdevicelink.api.audio;
+
+import android.content.Context;
+import android.media.AudioFormat;
+import android.media.MediaCodec;
+import android.media.MediaExtractor;
+import android.media.MediaFormat;
+import android.net.Uri;
+import android.os.Build;
+import android.support.annotation.NonNull;
+import android.support.annotation.RequiresApi;
+import android.util.Log;
+import com.smartdevicelink.api.audio.AudioStreamManager.SampleType;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+@SuppressWarnings("WeakerAccess")
+@RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
+public abstract class BaseAudioDecoder {
+    private static final String TAG = AudioDecoder.class.getSimpleName();
+
+    protected int targetSampleRate;
+    protected @SampleType int targetSampleType;
+
+    private int outputChannelCount;
+    private int outputSampleRate;
+    private @SampleType int outputSampleType;
+
+    private double lastOutputSample = 0;
+
+    private double lastOutputPresentationTimeUs = 0;
+    private double lastTargetPresentationTimeUs = 0;
+
+    protected MediaExtractor extractor;
+    protected MediaCodec decoder;
+
+    protected Uri audioSource;
+    protected Context context;
+    protected AudioDecoderListener listener;
+
+    public BaseAudioDecoder(Uri audioSource, Context context, int sampleRate, @SampleType int sampleType, AudioDecoderListener listener) {
+        this.audioSource = audioSource;
+        this.context = context;
+        this.listener = listener;
+
+        targetSampleRate = sampleRate;

```

Comment:
This class essentially assumes this value will never be 0, however, that value can certainly be supplied. We should add javadoc to the constructor to inform of the issue. I would also consider adding a default value of the supplied `sampleRate` was `<=0`

Reply:
Done, this value will be checked in AudioStreamManager.